data-otservbr-global/scripts/creaturescripts/quests/the_dream_courts/faceless_bane_step_positions.lua

```diff
@@ -0,0 +1,111 @@
+local walkedPositions = {}
+
+-- this variable controls the time to boss become immortal again, (after 1 minute, if dont die)
+local lastResetTime = os.time()
+
+-- reset sqm's / make the boss immortal again
+local function resetWalkedPositions()
+	if lastResetTime > os.time() then
+		return true
+	end
+	walkedPositions = {}
+	Game.setStorageValue(GlobalStorage.FacelessBaneStepsOn, 0)
+	lastResetTime = os.time() + (1 * 60)
+end
+
+-- energy effect to make the boss mortal
+local function sendEnergyEffect()
+	-- positions: pipes on room, that should be electrocuted
+	local pipePositions = {
```

Comment:
not that it will make a big difference, but if you leave the positions inside the function they will be related every time you call the function, I would leave them outside the function